### PR TITLE
Update Scala to 3.0.0 final, sbt to 1.5.2

### PR DESCRIPTION
**Pull request by lrytz**
... and migrate to sbt slash syntax

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]
LGTM!

File: project/Dependencies.scala [hunk 1]
```diff
@@ -29,15 +29,15 @@ object Dependencies {
...
+  val scala3Version = "3.0.0"
 
   val reactiveStreamsVersion = "1.0.3"
 
   val sslConfigVersion = "0.4.2"
 
   val scalaTestVersion = {
     if (getScalaVersion().startsWith("3.0")) {
-      "3.2.6"
+      "3.2.9"
```
Comment (raboof) [inline project/Dependencies.scala:40]: :+1:

File: project/Doc.scala [hunk 2]
```diff
@@ -153,15 +153,15 @@ object UnidocRoot extends AutoPlugin {
...
+        ScalaUnidoc / unidocProjectFilter := unidocRootProjectFilter(unidocRootIgnoreProjects.value),
+        JavaUnidoc / unidocProjectFilter := unidocRootProjectFilter(unidocRootIgnoreProjects.value),
+        ScalaUnidoc / apiMappings := (Compile / doc / apiMappings).value) ++
       UnidocRoot.CliOptions.genjavadocEnabled
         .ifTrue(
           Seq(
             // akka.stream.scaladsl.GraphDSL.Implicits.ReversePortsOps contains code that
-            // genjavadoc turns into (probably incorrect) Java code that in turn confuses the javadoc tool.
-            unidocAllSources in JavaUnidoc ~= { v =>
+            // genjavadoc turns into (probably incorrect) Java code turn / that confuses the javadoc tool.
```
Comment (raboof) [inline project/Doc.scala:163]:
```suggestion
            // genjavadoc turns into (probably incorrect) Java code that in turn confuses the javadoc tool.
```
:laughing:

File: project/JdkOptions.scala [hunk 3]
```diff
@@ -37,7 +37,7 @@ object JdkOptions extends AutoPlugin {
       targetSystemJdk,
       jdk8home,
       fullJavaHomes,
-      Seq("-target:jvm-1.8"),
+      Seq(if (Dependencies.getScalaVersion().startsWith("3.0")) "-Xtarget:8" else "-target:jvm-1.8"),
```
Comment (raboof) [inline project/JdkOptions.scala:40]: Ok :)

**Review by raboof — COMMENTED** [review]
Nice! One tiny tweak and then LGTM if CI agrees

File: project/JdkOptions.scala (hunk 3 above)
Comment (lrytz) [reply project/JdkOptions.scala:40]: This is for the `jdk8options` parameter; I think we could also just pass `Nil` here; when building on 8, `-target:8` is the default. What do you think?

Comment (raboof) [reply project/JdkOptions.scala:40]: IIRC `jdk8options` is for builds that should target jdk8 (while we build on jdk11+), so we probably need it. I was just surprised the option changed, but it's :+1: .

(yes, we'd ideally use `-release` instead, but #27079)

**Comment by raboof** [comment]
OK TO TEST

File: project/Doc.scala (hunk 2 above)
Comment (raboof) [reply project/Doc.scala:163]: ^ this was probably not intended though

File: project/JdkOptions.scala (hunk 3 above)
Comment (lrytz) [reply project/JdkOptions.scala:40]: In `private def selectOptions` there's `... else if (isJdk8) jdk8options else ...`; it seems the next parameter is for settings when building on 11.

File: project/Doc.scala (hunk 2 above)
Comment (lrytz) [reply project/Doc.scala:163]: yup 🙂

File: project/JdkOptions.scala (hunk 3 above)
Comment (raboof) [reply project/JdkOptions.scala:40]: ah, I misremembered. OK anyhow :)

**Comment by andreaTP** [comment]
LGTM

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**